ush/python/pygfs/task/analysis.py

```diff
@@ -0,0 +1,71 @@
+from pygw.task import Task
+from pygw.attrdict import AttrDict
+from pygw.template import Template, TemplateConstants
+from pygw.yaml_file import YAMLFile
+from pygw.file_utils import FileHandler
+import datetime as dt
+import logging
+import os
+
+crtmver = '2.3.0'
+
+
+class Analysis(Task):
+    """Parent class for GDAS tasks
+
+    The Analysis class is the parent class for all
+    Global Data Assimilation System (GDAS) tasks
+    directly related to peforming an analysis
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+
+    def initialize(self):
+        super().initialize()
+        # all analyses need to stage observations
+        obs_dict = self.get_obs_dict()
+        FileHandler(obs_dict).sync()
+
+    def configure(self):
+        """Compute additional variables and add them to the root configuration"""
+        super().configure()
+        # probably move some of below to the parent class eventually
+        self.PDY = self.config['PDY']
+        self.cyc = self.config['cyc']
+        self.current_cycle = dt.datetime.strptime(self.PDY + self.cyc, '%Y%m%d%H')
```

Comment:
These are universal enough they should probably be established in the parent class so all Tasks have them.

Reply:
True, will do.